### ISSUE TITLE
Update `forc index deploy` with option to not nuke old data

### ISSUE THREAD

**Issue by ra0x3** [FuelLabs/fuel-indexer#1127]
- If you run the service with `--replace-indexer`, everything related to the previous indexer with the same `namespace` and `identifier` gets deleted
- This is cool but makes debugging hard
  - Old blocks aren't as important as new blocks
  - If we wanna see how the indexer really performs, we want to use it on new blocks
  - However, with `--replace-indexer` you can't redeploy indexers without always having to start at the beginning
- We should add some flag `--keep-data` to `forc index deploy` that gets sent to the web endpoint
  - This flag would indicate that the user _does not_ want to delete data associated with their index
- This allows for redeploying resumable indexers

**Comment by ra0x3** [comment]
@lostman 
- As I'm thinking this over, I think:
  - The fact that `forc index deploy` with `--replace-indexer` removes all this stuff automagically is odd
  - `--replace-indexer` is set by the index service operator - the user has no idea what this flag is/does
  - We should make `forc index deploy` include explicit options for doing what the user wants
    - E.g., if the user wants to remove all old data then `forc index deploy --remove-data`
- `--replace-indexer` just signals that "If a user wants to replace their indexer, I will allow it"
  - But there are other things that go along with removing an indexer (e.g., removing data) that the user should decide


TLDR: User should opt-in to removing data with `--remove-data` (or something similar) rather than having everything removed by default, and having to use `--keep-data`. 

Thought?
